components/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import Page from './sub-components/Page';
+import ActionButton from './sub-components/ActionButton';
+import Dots from './sub-components/Dots';
+
+import pagination from './utils/pagination';
+
+const Wrapper = styled.nav`
+  align-items: center;
+  display: flex;
+  justify-content: center;
+  padding: 16px !important;
+`;
+
+const generateDotsKey = index => `dots-${index}`;
+
+const Pagination = ({
+  ariaLabel,
+  activePage,
+  activePageAriaLabel,
+  nextButtonText,
+  pageAriaLabel,
+  pageHref,
+  prevButtonText,
+  onPageClick,
+  totalPages,
+  ...props
+}) => {
+  const handlePageClick = page => e => {
+    if (!onPageClick) {
+      return undefined;
+    }
+
+    e.preventDefault();
+
+    if (page <= 0 || page > totalPages) {
+      return false;
+    }
+
+    return onPageClick(page);
+  };
+
+  const handleHref = page => {
+    if (!pageHref || onPageClick || page <= 0 || page > totalPages) {
+      return undefined;
+    }
+
+    return pageHref(page);
+  };
+
+  return (
+    <Wrapper aria-label={ariaLabel} {...props}>
+      <ActionButton
+        aria-disabled={activePage === 1}
+        disabled={activePage === 1}
```

Comment:
It's necessary to add `aria-disabled` prop when there is a `disabled` as well?

Reply:
Removed the `disabled` prop and using `aria-disabled` for the style.